docs/deprecation_policy.md

```diff
@@ -31,3 +37,9 @@ long process.
 By following the upstream support life cycles Synapse can ensure that its
 dependencies continue to get security patches, while not requiring system admins
 to constantly update their platform dependencies to the latest versions.
+
+For Rust, the situation is a bit different given that a) the Rust foundation
+does not generally support older Rust versions, and b) the library ecosystem
+generally bump their minimum support Rust versions frequently. In general,
+Synapse will unlikely ever require the absolute latest Rust version, but
```

Comment:
> In general, Synapse will unlikely ever require the absolute latest Rust version

This feels like it contradicts what is said above:

> For any given release the minimum required version may be bumped up to the latest released Rust version

Reply:
It's a hard one, as we want to reserve the right to do so, but will likely never actually do it except in exceptional circumstances? Not sure how better to frame it.

Reply:
I would say either remove the bit about Synapse's requirement from this sentence (ie turn it into something like "Because of this, introducing a formal policy is hard given the constraints of the ecosystem."), or reword the first half into something like "In practice, Synapse should only rarely require the absolute latest version of Rust". My issue with the current state of this sentence is that I read it as that in theory it's possible that requiring the latest version of Rust happens however in practice users should expect it to never happen, which isn't really the message we want to communicate.

Reply:
I've updated the wording a bit to make it a bit more balanced, maybe?

Reply:
Much better yep!